src/action.ts

```diff
@@ -48,33 +82,63 @@ export async function performAction(
 				logger.error(
 					`Found ${styledName} on ${styledTracker} - failed to inject, saving instead`
 				);
-				saveTorrentFile(
-					tracker,
-					getTag(searchee.name),
-					newMeta,
-					nonceOptions
-				);
+				saveTorrentFile(tracker, getTag(searchee.name), newMeta);
 				break;
 		}
 		return result;
 	} else {
-		saveTorrentFile(tracker, getTag(searchee.name), newMeta, nonceOptions);
+		saveTorrentFile(tracker, getTag(searchee.name), newMeta);
 		logger.info(`Found ${styledName} on ${styledTracker} - saved`);
 		return SaveResult.SAVED;
 	}
 }
 
-export async function performActions(searchee, matches, nonceOptions) {
+export async function performActions(searchee, matches) {
 	const results: ActionResult[] = [];
 	for (const { tracker, assessment } of matches) {
 		const result = await performAction(
 			assessment.metafile,
+			assessment.decision,
 			searchee,
-			tracker,
-			nonceOptions
+			tracker
 		);
 		results.push(result);
 		if (result === InjectionResult.TORRENT_NOT_COMPLETE) break;
 	}
 	return results;
 }
+
+function linkExact(oldPath: string, newPath: string) {
+	if (!newPath) {
+		return;
+	}
+	if (statSync(oldPath).isFile()) {
+		if (!existsSync(path.join(newPath, path.basename(oldPath)))) {
+			linkFile(oldPath, newPath);
```

Comment:
Shouldn't this be `linkFile(oldPath, path.join(newPath, path.basename(oldPath)));`

Reply:
absolutely, good catch